src/libman/Commands/InstallCommand.cs

```diff
@@ -200,9 +200,16 @@ private async Task<(string libraryId, ILibrary library)> ValidateLibraryExistsIn
                 {
                     // Found a group with an exact match.
                     string libraryId = libIds.First();
-                    ILibrary libraryToInstall = await ProviderCatalog.GetLibraryAsync(libraryId, cancellationToken);
 
-                    return (libraryId, libraryToInstall);
+                    try
+                    {
+                        ILibrary libraryToInstall = await ProviderCatalog.GetLibraryAsync(libraryId, cancellationToken);
+                        return (libraryId, libraryToInstall);
+                    }
+                    catch
+                    {
+                        // TO DO: Follow up with Prafull on what to o here
```

Comment:
@prafullbhosale added try/catch here since GetLibraryAsync can throw. Please review and see what needs to be done here.

Reply:
What does this throw and why?

---
In reply to: [195488891](https://github.com/aspnet/LibraryManager/pull/164#discussion_r195488891) [](ancestors = 195488891)

Reply:
GetLibraryAsync will throw if it is an invalid libaryId

---
In reply to: [195514626](https://github.com/aspnet/LibraryManager/pull/164#discussion_r195514626) [](ancestors = 195514626,195488891)

Reply:
Should we be throwing or returning null?  Is this truly an exceptional case?

---
In reply to: [195536200](https://github.com/aspnet/LibraryManager/pull/164#discussion_r195536200) [](ancestors = 195536200,195514626,195488891)